core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/StackDecorator.java

```diff
@@ -219,6 +220,15 @@ private Optional<AzureResourceGroup> getResourceGroupFromEnv(Stack stack, Detail
 
     private String getAvailabilityZoneFromEnv(DetailedEnvironmentResponse environment) {
         if (environment.getNetwork() != null && environment.getNetwork().getSubnetMetas() != null) {
+            String preferedSubnet = environment.getNetwork().getPreferedSubnetId();
+            if (StringUtils.isNotEmpty(preferedSubnet)) {
+                Optional<String> availabilityZone = environment.getNetwork().getSubnetMetas().entrySet().stream().filter(
+                        subnetMeta -> subnetMeta.getValue().getId().equals(preferedSubnet)
+                ).map(subnetMeta -> subnetMeta.getValue().getAvailabilityZone()).findFirst();
+                if (availabilityZone.isPresent()) {
+                    return availabilityZone.get();
```

Comment:
Can we get a log line here where the prefered subnet is gathered and what is that, please?

Reply:
yes